microprofile/WFLY-19591_MicroProfile_OpenAPI_4.0.adoc

```diff
@@ -0,0 +1,138 @@
+// issue that was used to track the previous feature.
+promotes:
+// This should be blank during initial development of a feature. It may be used
+// after the feature is completed if a subsequent issue is field to track promotion
+// of this feature to a higher stability level
+promoted-by:
+---
+= MicroProfile OpenAPI 4.0
+:author:            Paul Ferraro
+:email:             paul.ferraro@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+WildFly currently provides MicroProfile OpenAPI 3.1 support to applications.
+This feature request upgrades support for this specification to version 4.0 as part of version 7.0 of the MicroProfile platform specification.
+
+Most notably, this release of the MicroProfile OpenAPI specification generates documentation using version 3.1 of the OpenAPI specification.
+
+https://download.eclipse.org/microprofile/microprofile-open-api-4.0/microprofile-openapi-spec-4.0.html#release_notes_40
+
+=== User Stories
+
+This is feature is covered by existing user stories.
+
+== Issue Metadata
+
+* https://issues.redhat.com/browse/WFLY-19591[Parent issue]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-19588
+
+=== Affected Projects or Components
+
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+* Upgrade version of MicroProfile OpenAPI specification API to 4.0.
+* Upgrade version of SmallRye OpenAPI to 4.0.x
+
+=== Changed requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+=== Future Work
+
+N/A
+
+== Backwards Compatibility
+
+=== Default Configuration
+
+N/A
+
+=== Importing Existing Configuration
+
+N/A
+
+=== Deployments
+
+See https://download.eclipse.org/microprofile/microprofile-open-api-4.0/microprofile-openapi-spec-4.0.html#incompatible_changes_40[here] for the list of incompatible changes between versions 3.1 and 4.0 of the MicroProfile OpenAPI specification.
+
+=== Interoperability
+
+N/A
+
+== Implementation Plan
+
+The implementation of MicroProfile OpenAPI 4.0 is provided by SmallRye OpenAPI 4.0.x.
+The implemetation plan consist of simple version upgrade of the following components:
+
+* org.eclipse.microprofile.openapi API
+* org.eclipse.microprofile.openapi TCK
+* io.smallrye.openapi
+
+== Admin Clients
+
+N/A
+
+== Security Considerations
+
+N/A
+
+[[test_plan]]
+== Test Plan
+
+Version 4.0 of the MicroProfile OpenAPI TCK tests will be validated as part of the WildFly testsuite.
+WildFly-specific behavior and configuration are covered by existing tests within the WildFly testsuite.
+
+== Community Documentation
+
+References the MicroProfile OpenAPI and OpenAPI specifications need to be updated to reflect the current versions.
+ 
+== Release Note Content
+
+Support for the MicroProfile OpenAPI specification was updated from 3.1 to 4.0.
```

Comment:
Could you please mention the SmallRye OpenAPI `major.minor` version which will be used here in the suggested text for the Release Note? Something like [this](https://github.com/wildfly/wildfly-proposals/pull/649/files#diff-0c43b11664f7dcba3f6911ff130d4b6e7ef977d16d8777a9a9352bf2f89862b1R129-R130).
Readers (of notes, of course) will grab the relevant versions at first glance, this way.

Reply:
The version of the implementation library is not relevant to users - only the spec version.